// cbor/lib/cbor.js
'use strict'

exports.Commented = require('cbor/lib/commented')
exports.Diagnose = require('cbor/lib/diagnose')
exports.Decoder = require('cbor/lib/decoder')
exports.Encoder = require('cbor/lib/encoder')
exports.Simple = require('cbor/lib/simple')
exports.Tagged = require('cbor/lib/tagged')
exports.Map = require('cbor/lib/map')

exports.comment = exports.Commented.comment
exports.decodeAll = exports.Decoder.decodeAll
exports.decodeFirst = exports.Decoder.decodeFirst
exports.decodeAllSync = exports.Decoder.decodeAllSync
exports.decodeFirstSync = exports.Decoder.decodeFirstSync
exports.diagnose = exports.Diagnose.diagnose
exports.encode = exports.Encoder.encode
exports.encodeCanonical = exports.Encoder.encodeCanonical
exports.encodeOne = exports.Encoder.encodeOne
exports.decode = exports.Decoder.decodeFirstSync

exports.leveldb = {
  decode: exports.Decoder.decodeAllSync,
  encode: exports.Encoder.encode,
  buffer: true,
  name: 'cbor'
}

exports.hasBigInt = require('cbor/lib/utils').hasBigInt
