// fstream/fstream.js
exports.Abstract = require('fstream/lib/abstract.js')
exports.Reader = require('fstream/lib/reader.js')
exports.Writer = require('fstream/lib/writer.js')

exports.File = {
  Reader: require('fstream/lib/file-reader.js'),
  Writer: require('fstream/lib/file-writer.js')
}

exports.Dir = {
  Reader: require('fstream/lib/dir-reader.js'),
  Writer: require('fstream/lib/dir-writer.js')
}

exports.Link = {
  Reader: require('fstream/lib/link-reader.js'),
  Writer: require('fstream/lib/link-writer.js')
}

exports.Proxy = {
  Reader: require('fstream/lib/proxy-reader.js'),
  Writer: require('fstream/lib/proxy-writer.js')
}

exports.Reader.Dir = exports.DirReader = exports.Dir.Reader
exports.Reader.File = exports.FileReader = exports.File.Reader
exports.Reader.Link = exports.LinkReader = exports.Link.Reader
exports.Reader.Proxy = exports.ProxyReader = exports.Proxy.Reader

exports.Writer.Dir = exports.DirWriter = exports.Dir.Writer
exports.Writer.File = exports.FileWriter = exports.File.Writer
exports.Writer.Link = exports.LinkWriter = exports.Link.Writer
exports.Writer.Proxy = exports.ProxyWriter = exports.Proxy.Writer

exports.collect = require('fstream/lib/collect.js')
