// @hapi/hoek/lib/index.js
'use strict';

const internals = {};


module.exports = {
    applyToDefaults: require('@hapi/hoek/lib/applyToDefaults'),
    assert: require('@hapi/hoek/lib/assert'),
    Bench: require('@hapi/hoek/lib/bench'),
    block: require('@hapi/hoek/lib/block'),
    clone: require('@hapi/hoek/lib/clone'),
    contain: require('@hapi/hoek/lib/contain'),
    deepEqual: require('@hapi/hoek/lib/deepEqual'),
    Error: require('@hapi/hoek/lib/error'),
    escapeHeaderAttribute: require('@hapi/hoek/lib/escapeHeaderAttribute'),
    escapeHtml: require('@hapi/hoek/lib/escapeHtml'),
    escapeJson: require('@hapi/hoek/lib/escapeJson'),
    escapeRegex: require('@hapi/hoek/lib/escapeRegex'),
    flatten: require('@hapi/hoek/lib/flatten'),
    ignore: require('@hapi/hoek/lib/ignore'),
    intersect: require('@hapi/hoek/lib/intersect'),
    merge: require('@hapi/hoek/lib/merge'),
    once: require('@hapi/hoek/lib/once'),
    reach: require('@hapi/hoek/lib/reach'),
    reachTemplate: require('@hapi/hoek/lib/reachTemplate'),
    stringify: require('@hapi/hoek/lib/stringify'),
    wait: require('@hapi/hoek/lib/wait')
};
